src/webui/api/synccontroller.cpp

```diff
@@ -595,20 +586,27 @@ void SyncController::torrentPeersAction()
 qint64 SyncController::getFreeDiskSpace()
 {
     if (m_freeDiskSpaceElapsedTimer.hasExpired(FREEDISKSPACE_CHECK_TIMEOUT))
-    {
         invokeChecker();
-        m_freeDiskSpaceElapsedTimer.restart();
-    }
 
     return m_freeDiskSpace;
 }
 
-void SyncController::freeDiskSpaceSizeUpdated(qint64 freeSpaceSize)
+void SyncController::invokeChecker()
 {
-    m_freeDiskSpace = freeSpaceSize;
-}
+    if (m_isFreeDiskSpaceCheckerRunning)
+        return;
 
-void SyncController::invokeChecker() const
-{
-    QMetaObject::invokeMethod(m_freeDiskSpaceChecker, &FreeDiskSpaceChecker::check, Qt::QueuedConnection);
+    auto *freeDiskSpaceChecker = new FreeDiskSpaceChecker;
+    connect(freeDiskSpaceChecker, &FreeDiskSpaceChecker::checked, this, [this](const qint64 freeSpaceSize)
+    {
+        m_freeDiskSpace = freeSpaceSize;
+        m_isFreeDiskSpaceCheckerRunning = false;
+        m_freeDiskSpaceElapsedTimer.restart();
```

Comment:
`SyncController::getFreeDiskSpace()` does a restart also when calling `invokeChecker()`. I'll let you decide if you want to remove this restart or the other one.

Reply:
> `SyncController::getFreeDiskSpace()` does a restart also when calling `invokeChecker()`.

I don't know what you're talking about. I see only one restart.

Reply:
Oh damn it! I totally missed that you have already fixed the above.